sdk/storage/storage-file-share/test/fileserviceclient.spec.ts

```diff
@@ -380,4 +385,83 @@ describe("FileServiceClient", () => {
     assert.ok(typeof result.requestId);
     assert.ok(result.requestId!.length > 0);
   });
+
+  it("ListShares with deleted share", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+      }
+    }
+    assert.ok(found);
+  });
+
+  it("Undelete share positive", async function() {
+    let serviceClient: ShareServiceClient;
+    try {
+      serviceClient = getSoftDeleteBSU();
+    } catch (error) {
+      this.skip();
+    }
+    const shareClient = serviceClient.getShareClient(recorder.getUniqueName("share"));
+    await shareClient.create();
+    await shareClient.delete();
+
+    let found = false;
+    let shareDeleted: ShareItem | undefined;
+    for await (const share of serviceClient.listShares({ includeDeleted: true })) {
+      if (share.name == shareClient.name) {
+        found = true;
+        assert.ok(share.version);
+        assert.ok(share.deleted);
+        assert.ok(share.properties.deletedTime);
+        assert.ok(share.properties.remainingRetentionDays);
+
+        shareDeleted = share;
+      }
+    }
+    assert.ok(found);
+    assert.ok(shareDeleted);
+
+    // Await share to be deleted.
+    await delay(30 * 1000);
```

Comment:
Hi! Sorry to ask, but this seems brittle. Is there a way, or a plan, to make the delete operation a Long Running Operation? (using core-lro)

Reply:
Thanks for reviewing! As far as I know, there is no such plan. This might be reasonable, considering user might has little scenario to do something until delete finished.

---
In reply to: [475536974](https://github.com/Azure/azure-sdk-for-js/pull/10792#discussion_r475536974) [](ancestors = 475536974)